math/src/field/f64/tests.rs

```diff
@@ -125,6 +125,10 @@ fn element_as_int() {
     let v = u64::MAX;
     let e = BaseElement::new(v);
     assert_eq!(v % super::M, e.as_int());
+
+    let e1 = BaseElement::new(0);
+    let e2 = BaseElement::new(M);
+    assert_eq!(e1.as_int(), e2.as_int());
```

Comment:
Should we also add the following assertion?:
```Rust
assert_eq!(e1.as_int(), 0);
```
I think with the old code both `0` and `M` were getting reduced to `18446744069414584321` - so, I think we may need an extra check here.

Reply:
Very good point. It actually made me realize I had messed up the constant time branching, as I wasn't setting the binary mask to all 1s when the equality was satisfied. I've fixed it now, should be all good.